### PR TITLE
Drop quay creds overrides in build_deploy.sh

### PR DESCRIPTION
**Pull request by kurlov**
## Description

`build_deploy.sh` should rely on `QUAY_USER` and `QUAY_TOKEN`.
Also, image tag should be 7 chars long.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Documentation added if necessary
- [ ] CI and all relevant tests are passing

Not relevant

### CODE REVIEW

File: build_deploy.sh [hunk 1]
```diff
@@ -78,15 +73,15 @@ else
 fi
 
 # Push the image:
-echo "Quay.io user and token is set, will push images to $QUAY_ORG org"
+echo "Quay.io user and token is set, will push images to $QUAY_IMAGE_REPOSITORY"
```
Comment (porridge) [inline build_deploy.sh:76]: `$QUAY_IMAGE_REPOSITORY` has no default but is required.
Please add an `if [ -z` block like for the other `QUAY_...` vars.

Also, how do we set it in jenkins? 🤔

Comment (porridge) [reply build_deploy.sh:76]: I think we can follow this example:
```
data/services/cincinnati/cicd/ci-int/jobs.yaml:        build_deploy_script_path: env GRAPH_URL=https://api.stage.openshift.com:443/api/upgrades_info/v1/gra
ph ./dist/run_containerized_loadtesting.sh
```

Comment (kurlov) [reply build_deploy.sh:81]: We will set env var for it in Jenkins

File: build_deploy.sh [hunk 2]
```diff
@@ -29,12 +29,7 @@
 # the built images can be pushed to quay.io.
 
 # The version should be the short hash from git. This is what the deployment process expects.
```
Comment (porridge) [inline build_deploy.sh:31]:
```suggestion
# The version should be a 7-char hash from git. This is what the deployment process in app-interface expects.
```

File: build_deploy.sh [hunk 3]
```diff
@@ -54,12 +49,17 @@ cd "${LINK}"
...
 fi
 if [ -z "${QUAY_TOKEN}" ]; then
   echo "The quay.io push token hasn't been provided."
-  echo "Make sure to set the QUAY_USER_PASSWORD environment variable."
+  echo "Make sure to set the QUAY_TOKEN environment variable."
+  exit 1
+fi
+if [ -z "${QUAY_IMAGE_REPOSITORY}" ]; then
+  echo "The quay.io image repository hasn't been provided."
+  echo "Make sure to set the QUAY_IMAGE_REPOSITORY environment variable."
```
Comment (porridge) [inline build_deploy.sh:62]:
```suggestion
  echo "Make sure to set the QUAY_IMAGE_REPOSITORY environment variable to '<org>/<repo>'."
```

**Review by porridge — APPROVED** [review]
Just a couple of nitpicks.

**Pull request merged**